org/mozilla/jss/crypto/Policy.c

```diff
@@ -0,0 +1,54 @@
+#include "_jni/org_mozilla_jss_crypto_Policy.h"
+
+#include <nspr.h>
+#include <nss.h>
+#include <jni.h>
+
+jint
+nearest_power_of_two(jint value)
+{
+    for (jint exponent = 8; exponent < 20; exponent++) {
```

Comment:
@edewata On Ubuntu 19.10, `NSS_OptionGet(NSS_RSA_MIN_KEY_SIZE, &value)` returns `1023`. When this is passed to RSA key generation, it fails because it isn't a power of two. The minimum RSA key sizes supported is either 128 iirc, but the minimum DSA key size is 256, hence choosing `1 << 8 == 256` as the lower bound in case `value == 0`.

This correctly then moves 1023 -> 1024 and handles other scenarios.

On Fedora with `crypto-policies`, the value returned by `NSS_OptionGet(NSS_RSA_MIN_KEY_SIZE, &value)` is a power of two.

Reply:
LGTM.